Lib/inspect.py

```diff
@@ -2453,19 +2461,18 @@ def __init__(self, name, kind, *, default=_empty, annotation=_empty):
             raise ValueError('name is a required attribute for Parameter')
 
         if not isinstance(name, str):
-            raise TypeError("name must be a str, not a {!r}".format(name))
+            raise TypeError("name must be a str, not a {!s}".format(type(name).__name__))
 
         if name[0] == '.' and name[1:].isdigit():
             # These are implicit arguments generated by comprehensions. In
             # order to provide a friendlier interface to users, we recast
             # their name as "implicitN" and treat them as positional-only.
             # See issue 19611.
             if kind != _POSITIONAL_OR_KEYWORD:
-                raise ValueError(
-                    'implicit arguments must be passed in as {}'.format(
-                        _POSITIONAL_OR_KEYWORD
-                    )
-                )
+                msg = 'implicit arguments must be passed in as {!s} arguments not {!s} arguments'
+                msg.format(_PARAM_NAME_MAPPING[_POSITIONAL_OR_KEYWORD],
+                           _PARAM_NAME_MAPPING[self._kind])
```

Comment:
Really naive question but would it better to have more consistency between line 2471 and 2474, ie have something like:

    if kind != _POS:
        msg.format(_PARAM[_POS], _PARAM[kind])

or:

    if self._kind != _POS:
        msg.format(_PARAM[_POS], _PARAM[self._kind])

In any case, I love the idea of adding more details in the error message :)

Reply:
@SylvainDe Yeah, that might be better :)